pkg/api/tests.go

```diff
@@ -210,9 +210,8 @@ func BuildTestsResults(dbc *db.DB, release, period string, collapse, includeOver
 	}
 
 	testReports := make([]apitype.Test, 0)
-	// FIXME: Add test id to matview, for now generate with ROW_NUMBER OVER
 	processedResults := dbc.DB.Table("(?) as results", rawQuery).
-		Select(`ROW_NUMBER() OVER() as id, watchlist, name, jira_component, jira_component_id,` + variantSelect + query.QueryTestSummarizer).
+		Select(`id, watchlist, name, jira_component, jira_component_id,` + variantSelect + query.QueryTestSummarizer).
```

Comment:
This is likely worth additional review.  I'm wondering if there was a reason id wasn't used before and if this impacts other parts of the UI like the tests report.

Reply:
Could select id as test_id and then keep the existing ROW_NUMBER() OVER() statement.  Don't know that we need it but if we wanted to keep the query as it was and just add in the test id.

Reply:
from the deleted FIXME comment and the state of the matview, i gathered this was written at a point where the test id didn't exist in the view, and so ROW_NUMBER was just a stand-in so there would be _something_, with the intent to go back and update the matview and the query. well, the matview got updated with a test id, but the query never did until this.

Reply:
Looks like apitype.Test expects ID to be a float64: https://github.com/openshift/sippy/blob/cc0ba0efaad03249186db2b1e0b9fbfac41ef202/pkg/apis/api/types.go#L535

Reply:
/hold
the tests report is using the "synthetic" id and it gets the table wrong if results repeat the same id. that seems like an abuse of `Test.ID` to me but i suppose we can work around it.

no idea what the float64 cast is for but we can leave Test.ID to its current purpose and look up the real test id in a new field.